kubelet/datadog_checks/kubelet/cadvisor.py

```diff
@@ -0,0 +1,177 @@
+        Nested metric names are flattened: memory/usage -> memory.usage
+        """
+        if depth >= self.max_depth:
+            self.log.warning('Reached max depth on metric=%s' % metric)
+            return
+
+        if isinstance(dat, numbers.Number):
+            # Pod level metric filtering
+            is_pod_metric = False
+            if self.pod_level_metrics and any([fnmatch(metric, pat) for pat in self.pod_level_metrics]):
+                is_pod_metric = True
+            if is_pod_metric != is_pod:
+                return
+
+            # Metric submission
+            if self.enabled_rates and any([fnmatch(metric, pat) for pat in self.enabled_rates]):
+                self.rate(metric, float(dat), tags)
+            elif self.enabled_gauges and any([fnmatch(metric, pat) for pat in self.enabled_gauges]):
+                self.gauge(metric, float(dat), tags)
+
+        elif isinstance(dat, dict):
+            for k, v in dat.iteritems():
+                self._publish_raw_metrics(metric + '.%s' % k.lower(), v, tags, is_pod, depth + 1)
+
+        elif isinstance(dat, list):
+            self._publish_raw_metrics(metric, dat[-1], tags, is_pod, depth + 1)
+
+    def _update_container_metrics(self, instance, subcontainer, pod_list, container_filter):
+        is_pod = False
+        in_static_pod = False
+        cid = subcontainer.get('id')
+        pod_uid = subcontainer.get('labels', []).get('io.kubernetes.pod.uid')
+        k_container_name = subcontainer.get('labels', []).get('io.kubernetes.container.name')
+
+        # We want to collect network metrics at the pod level
+        if k_container_name == "POD" and pod_uid:
+            is_pod = True
+
+        # FIXME we are forced to do that because the Kubelet PodList isn't updated
+        # for static pods, see https://github.com/kubernetes/kubernetes/pull/59948
+        pod = get_pod_by_uid(pod_uid, pod_list)
+        if pod is not None and is_static_pending_pod(pod):
+            in_static_pod = True
+
+        # Let's see who we have here
+        if is_pod:
+            tags = tags_for_pod(pod_uid, True)
+        elif in_static_pod and k_container_name:
+            tags = tags_for_docker(cid, True)
+            tags += tags_for_pod(pod_uid, True)
+            tags.append("kube_container_name:%s" % k_container_name)
+        else:  # Standard container
+            if container_filter.is_excluded(cid):
+                self.log.debug("Filtering out " + cid)
+                return
+            tags = tags_for_docker(cid, True)
+
+        if not tags:
+            self.log.debug("Subcontainer {} doesn't have tags, skipping.".format(cid))
+            return
+        tags = list(set(tags + instance.get('tags', [])))
+
+        stats = subcontainer['stats'][-1]  # take the latest
+        self._publish_raw_metrics(NAMESPACE, stats, tags, is_pod)
+
+        if is_pod is False and subcontainer.get("spec", {}).get("has_filesystem") and stats.get('filesystem', []) != []:
```

Comment:
nit: what about doing this instead:
```python
if is_pod is False and subcontainer.get("spec", {}).get("has_filesystem") and stats.get('filesystem'):
```
It doesn't create two empty lists to just compare if it's not `None`.